MoreLinq.Test/TriplewiseTest.cs

```diff
@@ -0,0 +1,49 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2008-2011 Jonathan Skeet. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class TriplewiseTest
+    {
+        [Test]
+        public void TriplewiseIsLazy()
+        {
+            new BreakingSequence<object>().Triplewise(BreakingFunc.Of<object, object, object, int>());
+        }
+
+        [TestCase(0)]
+        [TestCase(1)]
+        [TestCase(2)]
+        public void TriplewiseWithSequenceShorterThanTwo(int count)
+        {
+            var source = Enumerable.Range(0, count);
+            var result = source.Triplewise(BreakingFunc.Of<int, int, int, int>());
+
+            Assert.That(result, Is.Empty);
+        }
+
+        [Test]
+        public void TriplewiseWideSourceSequence()
+        {
+            var result = new[] { "a", "b", "c", "d", "e" }.Triplewise((x, y, z) => x + y + z);
+            result.AssertSequenceEqual("abc", "bcd", "cde");
+        }
```

Comment:
Add tests cases mechanics can allow you to merge `TriplewiseWithSequenceShorterThanTwo` and `TriplewiseWideSourceSequence`.

Here an example:
https://github.com/morelinq/MoreLINQ/blob/5ad0a6528a6d7708189518396026fd0c05df65c1/MoreLinq.Test/ZipLongestTest.cs#L31-L52